testutils/src/main/scala/za/co/absa/enceladus/testutils/datasetComparison/ComparisonJob.scala

```diff
@@ -92,7 +91,7 @@ object ComparisonJob {
 
     if (errorsPresent) {
       if (cmd.keys.isDefined) {
-        handleKeyBasedDiff(cmd, expectedDf, expectedMinusActual, actualMinusExpected)
+        handleKeyBasedDiff(cmd.keys.get, cmd.outPath, expectedMinusActual, actualMinusExpected)
```

Comment:
Why not replace this `if/else` with pattern matching? `.isDefined` + `.get` is sort of an anti-pattern:
```
cmd.keys match {
  case Some(keys) => handleKeyBasedDiff(keys, cmd.outPath, expectedMinusActual, actualMinusExpected)
  case None       => handleSeparateDiff(...)
}
```